api/storage_service.cc

```diff
@@ -312,6 +312,15 @@ void set_repair(http_context& ctx, routes& r, sharded<repair_service>& repair) {
                 });
     });
 
+    ctx.db.local().get_builtin_routine_registry().register_routine("keyspace_repair",
+            db::make_single_value_routine<int>("repair", [&ctx, &repair](std::unordered_map<sstring, sstring>&& val) {
+        auto ks = val.contains("keyspace")
+            ? validate_keyspace(ctx, val["keyspace"])
+            : throw bad_param_exception("No keyspace specified");
+        val.erase("keyspace");
+        return repair_start(repair, ks, val);
```

Comment:
Is this synchronous or asynchronous? Does it return when repair completes, or just started?

It should be synchronous, so scripts don't have to poll.

Reply:
It returns when repair has been started